docs/development/new-kubernetes-version.md

```diff
@@ -0,0 +1,122 @@
+# Adding Support For A New Kubernetes Version
+
+This document describes the steps needed to perform in order to confidently add support for a new Kubernetes **minor** version.
+
+> ⚠️ Typically, once a minor Kubernetes version `vX.Y` is supported by Gardener then all patch versions `vX.Y.Z` are also automatically supported without any required action.
+This is because patch versions do not introduce any new feature or API changes, so there is nothing that needs to be adapted in `gardener/gardener` code.
+
+The Kubernetes community release a new minor version roughly every 4 months.
+Please refer to the [official documentation](https://kubernetes.io/releases/release/) about their release cycles for any additional information.
+
+Shortly before a new release an "umbrella" issue should be opened which is used to collect the required adaptations and to track the work items.
+For example, [#5102](https://github.com/gardener/gardener/issues/5102) can be used as a template for the issue description.\
```

Comment:
If you want, you can mention that upgrading the kubernetes dependencies is typically done as separate tasks.